Update install-prereqs-arch.sh

igoutta opened the pull request:
- Added the much needed opencv-python package so that MAVProxy's console and map are finally displayed.
- Automatic and improved way to add the venv and path variables to the shell config file, whether you use bash or zsh.
- Corrected the AUTOPILOT_TOOLS directory with a reliable source for obtaining it.
- Removed the annoying `set -x` which filled the whole terminal with the result of every command.

igoutta commented:
I deleted the QUIET option because it was completely unused. Funny because the `set -x` was really loud.

rmackay9 commented:
Hi @igoutta,

thanks for this.  Two small things:

1. could you update the commit title to start with "Tools: "?  If you peek at our commit list you'll see we always do this because it makes backporting easier
2. could you rebase on master to remove the merge commit?  We don't accept merge commits into the master branch again because it makes backporting easier when we don't have them

igoutta commented:
Hi, @rmackay9.

I'm happy to help. I rewrote the whole script without changing or adding any packages except the ones mentioned above.

The review of changes:

- Upgraded algorithm (with help of Deepseek, ngl).
- QUIET mode came back and it's actually quiet.
- No more change directory to do stuff.
- Lots of verifications to ensure every step is properly functioning at the end.
- Some advices at the end.

I have tested this script a lot in this computer. I tried to be at resourceful as I could, so I really wish anyone with Arch could enjoy using it. I personally disagree with the handling of the virtual enviroment but that's another history.

I noticed some path errors which I corrected, I guess from the translation of Ubuntu into Arch. The use of the script directory in the git command when you really need the root of the repository. The function of prompt returning 1, which basically means a fail of the script. Now it return 0 but still do the intented function.

I hope it works this time. Sincerely.

rmackay9 commented:
I wonder if @khancyr is perhaps the best one to review this..

peterbarker commented on Tools/environment_install/install-prereqs-arch.sh:
```diff
@@ -1,145 +1,185 @@
+echo "Installing Python packages..."
+PIP_OPTS=(-U --no-warn-script-location)
+"$QUIET" && PIP_OPTS+=(-q)
+"$VENV_DIR"/bin/python3 -m pip install "${PIP_OPTS[@]}" --upgrade pip
+"$VENV_DIR"/bin/python3 -m pip install "${PIP_OPTS[@]}" "${PYTHON_PKGS[@]}"
```
This looks actively bad; If the `python3` in your path isn't from that venv then something seriously bad is going on.

igoutta replied:
If you read the code there I'm positive you will see that this code execute the Python binary from the virtual enviroment, especially because if there is no folder or the install of the virtual envoriment goes wrong then the code exit. Unless you haven't run any python script from any IDE, from where I get the idea.

peterbarker reviewed (changes requested):
I am seriously not a fan of a complete rewrite here - including functional changes.

Please do this in more incremental steps, and bear in mind this script isn't stand-alone, it's part of a collection of scripts which are all much-of-a-muchness and need to be maintained by the same people.

igoutta commented on Tools/environment_install/install-prereqs-arch.sh:
```diff
@@ -1,145 +1,185 @@
 #!/usr/bin/env bash
-set -e
-set -x
```
That's awful. It's confusing, And does not gives you the debugging you think it give you. If bloated into the extremis.

igoutta commented on Tools/environment_install/install-prereqs-arch.sh:
```diff
@@ -1,145 +1,185 @@
 
-PYTHON_PKGS="future lxml pymavlink MAVProxy pexpect argparse matplotlib pyparsing geocoder pyserial empy==3.3.4 dronecan packaging setuptools wheel"
+# Development Packages
+BASE_PKGS=(base-devel gcc ccache git wget gsfonts tk)
+SITL_PKGS=(python-pip python-setuptools python-wheel python-numpy python-scipy opencv python-wxpython)
+PX4_PKGS=(lib32-glibc zip zlib ncurses)
+PYTHON_PKGS=(future lxml pymavlink MAVProxy opencv-python pexpect argparse matplotlib pyparsing geocoder pyserial empy==3.3.4 dronecan packaging setuptools wheel)
```
I prefer it. It's a matter of choice.

igoutta commented on Tools/environment_install/install-prereqs-arch.sh:
```diff
@@ -1,145 +1,185 @@
+    [[ $REPLY =~ ^[Yy]$ ]]
+}
 
-function maybe_prompt_user() {
-    if $ASSUME_YES; then
-        return 0
-    else
-        read -p "$1"
-        if [[ $REPLY =~ ^[Yy]$ ]]; then
```
Originally you didn't have any real output in the quiet option for Arch but you are complaining about reworking the function about asking in order to have readable code. There is nothing I can do about it.

igoutta commented:
> I am seriously not a fan of a complete rewrite here - including functional changes.
> 
> Please do this in more incremental steps, and bear in mind this script isn't stand-alone, it's part of a collection of scripts which are all much-of-a-muchness and need to be maintained by the same people.

I'm simply trying to support a script that didn't even gives the Arch user a functional installation. The original script lacked of so many essential things, starting from the OpenCV library in the Python packages, needed to have GUI for MavProxy. The zero use of the quiet option, and `set -x` literally bloating the screen without a real debug. Lack of self-awareness about the used shell. 

If you want to make changes make them, perharps checking in the full extend if the Python binary from the virtual envoriment exists, or is not corrupt. Things that takes time or a discussion to understand if they're needed, because they're the product of  another tools used in the script.

peterbarker commented on Tools/environment_install/install-prereqs-arch.sh:
```diff
@@ -1,145 +1,185 @@
+echo "Initializing repository submodules..."
+GIT_OPTS=(--init --recursive)
+"$QUIET" && GIT_OPTS+=(--quiet)
+git -C "${SCRIPT_DIR%/Tools/environment_install}" submodule update "${GIT_OPTS[@]}"
```
The original formulation stops being from having to go and look up what `-C` and `%` do.  And no, that's no on them for being stupid, this is them having other things on their mind.  Simpler is better.

peterbarker reviewed (changes requested):
I'm not saying at all that your script here isn't better from a technical standpoint, making use of all sorts of things that make life nicer in shell scripting.

I'm saying that "simpler is better" sometimes, and that familiarity is good.

timtuxworth commented:
I notice this claims to 

> Added the much needed opencv-python package so that MAVProxy's console and map are finally displayed.

I'm having a problem where console and map are not displayed on my WSL2 22.04 installation, is this going to solve that?

igoutta commented:
> Hi @igoutta,
> 
> thanks for this. Two small things:
> 
> 1. could you update the commit title to start with "Tools: "?  If you peek at our commit list you'll see we always do this because it makes backporting easier.  To be clear, i mean the commit title, not the PR title.
> 2. could you rebase on master to remove the merge commit?  We don't accept merge commits into the master branch again because it makes backporting easier when we don't have them

Finally changed that commit xd

igoutta commented:
> I notice this claims to
> 
> > Added the much needed opencv-python package so that MAVProxy's console and map are finally displayed.
> 
> I'm having a problem where console and map are not displayed on my WSL2 22.04 installation, is this going to solve that?

I could try to check that script as well. Last time I tried, I worked well but I don't really remember well. This weekend I could make some time.
